File: akkaStreamsCustomStage/src/main/java/com/lightbend/java/modelserver/queryablestate/QueriesAkkaHTTPResource.java
package com.lightbend.java.modelserver.queryablestate;

import akka.http.javadsl.marshallers.jackson.Jackson;
import akka.http.javadsl.server.AllDirectives;
import akka.http.javadsl.server.Route;
import com.lightbend.java.modelserver.modelserver.ReadableModelStore;

public class QueriesAkkaHTTPResource extends AllDirectives {

    private ReadableModelStore reader;

    public QueriesAkkaHTTPResource(ReadableModelStore reader){
        this.reader = reader;
    }

    public Route createRoute() {
        return route(
                path("state", () ->
                    get(() -> completeOK(reader.getCurrentServingInfo(), Jackson.marshaller())))
        );
    }

}

File: akkaStreamsCustomStage/src/main/scala/com/lightbend/scala/modelServer/queriablestate/QueriesAkkaHttpResource.scala
package com.lightbend.scala.modelServer.queriablestate

import akka.http.scaladsl.server.Route
import akka.http.scaladsl.server.Directives._
import com.lightbend.scala.modelServer.model.ModelToServeStats
import com.lightbend.scala.modelServer.modelServer.ModelStateStore
import de.heikoseeberger.akkahttpjackson.JacksonSupport

object QueriesAkkaHttpResource extends JacksonSupport {

  def storeRoutes(predictions: ModelStateStore): Route =
    get {
      path("state") {
        val info: ModelToServeStats = predictions.getCurrentServingInfo
        complete(info)
      }
    }
}
